perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -154,6 +212,96 @@ def CheckPrerequisites():
     data.ResourcePath(FLAGS.cassandra_stress_profile)
 
 
+def CheckMetadata(metadata):
+  """Verify that metadata is valid.
+
+  Args:
+    metadata: dict. Contains metadata for this benchmark.
+  """
+  if metadata['command'] in PRELOAD_REQUIRED:
+    if metadata['pop_size'] > metadata['num_preload_keys']:
+      raise errors.Benchmarks.PrepareException(
+          'For %s modes, number of preloaded keys must be larger than or '
+          'equal to population size.', PRELOAD_REQUIRED)
+
+
+def SetupMetadata(benchmark_spec):
```

Comment:
What is the motivation for attaching this to the benchmark spec? We should avoid adding arbitrary data to objects.